### PR TITLE
plugin handler: process elf files only if base is specified

### PR DESCRIPTION
**Pull request by cjp256**
ELF files are always processed when `type: app`, where `app`
is the default type.

This causes an issue when snapd is building without specifying
the type, as it gets treated as `type: app` and its files
are processed like typical app snaps.  The schema currently
allows for this because it fails to set the default type `app`
in time for validation.

Since snapd does not have a base, check for the existence of the
base before attempting to process ELF files.  This will prevent
the error reported in LP #1857019 and allow snapd to build without
a type specified.

Once snapd moves to `type: snapd`, we can fix the schema to
ensure that the invalid configuration that led to this error
does not pass verification.

LP #1857019
LP #1862642

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/internal/pluginhandler/__init__.py [hunk 1]
```diff
@@ -815,7 +815,7 @@ def _do_prime(self) -> None:
         snap_files, snap_dirs = self.migratable_fileset_for(steps.PRIME)
         _migrate_files(snap_files, snap_dirs, self.stagedir, self.primedir)
 
-        if self._snap_type == "app":
+        if self._snap_type == "app" and self._base:
```
Comment (sergiusens) [inline snapcraft/internal/pluginhandler/__init__.py]: I don't know why, but lately I prefer `is not None` for explicit `None` checks.
This is fine though, just thought I'd mention it.

Comment (cjp256) [reply snapcraft/internal/pluginhandler/__init__.py]: Done

**Comment by cjp256** [comment]
Added spread test.

**Comment by anonymouse64** [comment]
I'm a little late to report this, but I built https://github.com/snapcore/snapd/pull/7904 with this PR (the `--channel=edge/pr-2926` thing is really cool :+1: ) and it worked fine, so we are unblocked on the snapd side as soon as this lands in the candidate channel.

**Pull request merged**